src/Core/Logger/LoggerManager.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+// Copyright (C) 2010-2024, the Friendica project
+// SPDX-FileCopyrightText: 2010-2024 the Friendica project
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+declare(strict_types=1);
+
+namespace Friendica\Core\Logger;
+
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\Logger\Capability\LogChannel;
+use Friendica\Core\Logger\Factory\LoggerFactory;
+use Friendica\Core\Logger\Type\ProfilerLogger;
+use Friendica\Core\Logger\Type\WorkerLogger;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Psr\Log\NullLogger;
+
+/**
+ * Manager for the core logging instances
+ */
+final class LoggerManager
+{
+	/**
+	 * Workaround: $logger must be static
+	 * because Dice always creates a new LoggerManager object
```

Comment:
I thought Dice could be configured to return a singleton object only created once?

Reply:
Yes, that's true if we call `$dice->create()`. But for `LoggerInterface` we registered a factory and this factory is not created ones, but created always new. That's a flaw of Dice, so I had to introduce this workaround.